File: code/724.find-pivot-index.php
<?php
/*
 * @lc app=leetcode id=724 lang=php
 *
 * [724] Find Pivot Index
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function pivotIndex($nums) {
        $leftSum = 0;
        $rightSum = array_sum($nums) - $nums[0];
        if ($leftSum === $rightSum) return 0;

        for ($i = 1; $i < count($nums); $i++) {
            $leftSum += $nums[$i - 1];
            $rightSum -= $nums[$i];

            if ($leftSum === $rightSum) return $i;
        }

        return -1;
    }
}
// @lc code=end

File: code/724.find-pivot-index.go
package code

/*
 * @lc app=leetcode id=724 lang=golang
 *
 * [724] Find Pivot Index
 */

// @lc code=start
func pivotIndex(nums []int) int {
	sumLeft := 0
	sumRight := 0
	for i := 1; i < len(nums); i++ {
		sumRight += nums[i]
	}

	if sumLeft == sumRight {
		return 0
	}

	for i := 1; i < len(nums); i++ {
		sumLeft += nums[i-1]
		sumRight -= nums[i]

		if sumLeft == sumRight {
			return i
		}
	}

	return -1
}

// @lc code=end
